src/sentry/integrations/opsgenie/client.py

```diff
@@ -37,11 +39,27 @@ def get_alerts(self, limit: int | None = 1) -> object | None:
         path = f"/alerts?limit={limit}"
         return self.get(path=path, headers=self._get_auth_headers())
 
+    def _get_workflow_urls(self, group, rules):
+        organization = group.project.organization
+        workflow_urls = []
+        for rule in rules:
+            # fetch the workflow_id from the rule.data
+            workflow_id = rule.data.get("workflow_id")
+            assert workflow_id is not None
+            workflow_urls.append(
+                organization.absolute_url(create_link_to_workflow(organization.id, workflow_id))
+            )
+        return workflow_urls
+
     def _get_rule_urls(self, group, rules):
         organization = group.project.organization
         rule_urls = []
         for rule in rules:
-            path = f"/organizations/{organization.slug}/alerts/rules/{group.project.slug}/{rule.id}/details/"
+            rule_id = rule.id
+            if features.has("organizations:workflow-engine-trigger-actions", organization):
+                rule_id = rule.data.get("legacy_rule_id")
```

Comment:
Do we want some assertion that this is defined? It shouldn't happen, but creating a link with `None/details` seems worse.